moabb/contexts/base.py

```diff
@@ -44,7 +46,29 @@ def __init__(self, datasets, pipelines):
                 raise(ValueError("pipelines must only contains Pipelines instance"))
         self.pipelines = pipelines
 
-
+    @abstractmethod
+    def score(self, clf, X, y, groups, n_jobs=1):
+        '''
+        Return score 
+        '''
+        pass
+
+    @abstractmethod
+    def prepare_data(self, dataset, subjectlist):
+        '''
+        Given dataset, fetch data from subjects
+
+        Parameters:
+            dataset:       Dataset instance
+            subjectlist:   List of ids **(strings? numbers?)** for subjects
+
+        Output:
+            X:       ndarray (trials, channels, timepoints) of data
+            y:       ndarray (trials,) **1 or 2d** of labels
+            groups:  ndarray (trials,) specifying which subject each trial belongs to
```

Comment:
Given that there is a list of subjects this is an issue, right? That needs to be preserved in X an y as a general rule, I think? So we could alter the signature in all classes such that X and y and groups are lists where each member corresponds to the data of the corresponding member of subjectlist?